protocol/0062-SPAM-spam_protection.md

```diff
@@ -60,8 +60,7 @@ For Sweetwater, the policies we enforce are relatively simple:
  pre_block rejected, if there are `<num_votes>` or more votes on the same proposal in the blockchain in this epoch, and
  post_block rejected, if there are `<num_votes>` or more on the same proposal in the blockchain plus earlier in the current block.
 
-- Any tokenholder that had more than 50% if its votes post-rejected is banned for 4 epochs, and all its votes are immediately 
-  rejected. 
+- Any tokenholder that had more than 50% if its post-rejected is banned for max (30 seconds, 1/48 of an epoch) or until the next epoch starts, and all its votes (and only those, no trading related transactions) are immediately rejected. E.g. if the epoch duration is 1 day, then the ban period is 30 minutes. If however the epoch is 10 seconds, then the ban period is 30 seconds (or until the start of the next epoch). The test for 50% of the votes is repeated once the next vote is post-rejected, so a it is possible for a violating party to get banned quite quickly again.
```

Comment:
A couple of comments:
1. this is not only for voting, it's applicable for proposals, delegation, etc. 
2. what happens once the party gets unbanned? lets say they have voted already 3 times time epoch and the above applied to them so they got banned, half an hour passes and ban is lifted. Can they vote again for a given proposal to which they've already voted 3 times or should they be rejected?

Reply:
1. True, governance-related transactions would be more precise.
2. The vote count for that epoch stays, so the fourth vote will now be rejected; it will not cause another ban, as the state of the validators should have caught up now, so spam-protection should reject that vote before it enters a block. If the account manages to sneak some transaction through that needs to be post-rejected (e.g., through a vote on another issue), this post rejection can immediately trigger a new ban through the 50% rule.

Reply:
excellent, thanks.